### DISCUSSION TITLE
No luck in WSL

### DISCUSSION THREAD

**Discussion by Ks4four** [regolith-linux/regolith-desktop#1040] [Q&A] [answered]
I wonder if it is possible to install regolith-desktop in WSL2?
Using Vcxsrv, run sudo regolith-session got this:

![image](https://github.com/user-attachments/assets/3d82cb3b-6bbb-46af-a5b8-02fce240a6e9)

Any suggestions?

**Comment by kgilmer** [comment, marked as answer]
Interesting @Ks4four , well there has been a lot of work to clean up and modernize how Regolith starts up recently.  If you didn't try with 3.2 (in beta), I'd suggest trying that.  To install the beta, you can change the section of the Regolith `apt` URL from "release-3_1" to "testing", then update or install the regolith packages as described on the install page.

Also, once you've installed the beta, maybe try both x11 and wayland sessions.  I'd be interested to know how it goes!

**Reply by Ks4four** [reply]
Updated to the testing, and I found [this line](https://github.com/Seluj53/WSL-Regolith/blob/4d2bf85d96cd10c4e6c6b12306c016763e8a772b/windows/wslup.vbs#L6) works:
```
vcxsrv.exe :0 -screen 0 @1 -ac +xinerama -engine 1 -nodecoration -wgl -keyhook -clipboard
```

Tested on x11 using `sudo -u username env XDG_SESSION_TYPE=X11 regolith-session-x11`.
~~But I found that I'm always in root, which makes some apps would not open (for example Anki) via the launcher.  How to make it stay on normal user when open the session?~~

And how to open a wayland session, by the way?
![Regolith-desktop in WSL](https://github.com/user-attachments/assets/d19c5c24-820e-42d0-b455-5d37e9590f3f)

**Reply by kgilmer** [reply]
Looks like `i3-swap-focus` expects an env variable that isn't being set.  Here's what mine is (on Wayland):

```
$ echo $XDG_RUNTIME_DIR 
/run/user/1000
```

I'm not familiar with WSL.  Typically it's selected via the display manager (gdm3, lightdm).  You might have luck w/ running the exec line here from where you are starting the X11 session: https://github.com/regolith-linux/regolith-session/blob/main/usr/share/wayland-sessions/regolith-wayland.desktop

**Reply by Ks4four** [reply]
I don't have the desktop entry for now, weird.
WSL is on Ubuntu 22.04, but I found [an issue](https://github.com/regolith-linux/regolith-desktop/issues/913#issuecomment-1724989189) that sway doesn't officially support proprietary drivers so I am lazy to trying the wayland. X11 session is good enough. Thanks!
